### PR TITLE
Added missing invoke for AppDomain.UnhandledException

### PR DESCRIPTION
**Pull request by AlexDoeringEnvision**
#961

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtimelab?pullRequest=1134) <br/>All CLA requirements met.

**Review by jkotas — APPROVED** [review]
Nice. Thank you!

File: src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs [hunk 1]
```diff
@@ -223,6 +223,26 @@ private static void OnFirstChanceExceptionViaClassLib(object exception)
...
+            if (pOnUnhandledExceptionFunction == IntPtr.Zero)
+            {
+                return;
+            }
+
+            try
+            {
+                ((delegate*<object, void>)pOnUnhandledExceptionFunction)(exception);
+            }
+            catch when (true)
```
Comment (Suchiman) [inline src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs:240]: > warning CS7095: Filter expression is a constant 'true', consider removing the filter
```suggestion
            catch
```

Comment (jkotas) [reply src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs:240]: `catch when (true)` is used in the rest of the file, so I think it is fine to use here as well for consistency.

We use to have a mode where the core runtime was a different type universe from the app (`System.Object` here was not the apps `System.Object`). This is left-over from that mode. `catch when (true)` was needed here to catch all exceptions. Regular `catch` would only catch the exceptions from this type universe.

**Pull request merged**